include/alpaka/core/Common.hpp

```diff
@@ -37,6 +37,19 @@
 #    define ALPAKA_FN_HOST
 #endif
 
+//-----------------------------------------------------------------------------
+//! All functions marked with ALPAKA_FN_ACC or ALPAKA_FN_HOST_ACC that are exported to / imported from different
+//! translation units have to be attributed with ALPAKA_FN_EXTERN.
+//!
+//! Usage:
+//! ALPAKA_FN_ACC ALPAKA_FN_EXTERN auto add(std::int32_t a, std::int32_t b) -> std::int32_t;
+#ifdef ALPAKA_ACC_SYCL_ENABLED
+#    define ALPAKA_FN_EXTERN SYCL_EXTERNAL
+#else
+#    define ALPAKA_FN_EXTERN
+#endif
```

Comment:
Why? Is that required from the SYCL spec? Could you also please document the reason?

Reply:
I will add a comment for this. The source is the SYCL 2020 spec, section 5.10.1:

> The default behavior in SYCL applications is that all the definitions and declarations of the functions and member functions are available to the SYCL compiler, in the same translation unit. When this is not the case, all the symbols that need to be exported to a SYCL library or from a C++ library to a SYCL applica­tion need to be defined using the macro: `SYCL_EXTERNAL`.